mcs/class/corlib/Test/System.Reflection/AssemblyTest.cs

```diff
@@ -55,23 +55,31 @@ namespace MonoTests.System.Reflection
 	[TestFixture]
 	public class AssemblyTest
 	{
-		static string TempFolder = Path.Combine (Path.GetTempPath (),
+		static string BaseTempFolder = Path.Combine (Path.GetTempPath (),
 			"MonoTests.System.Reflection.AssemblyTest");
+		static string TempFolder;
 
 		[SetUp]
 		public void SetUp ()
 		{
-			while (Directory.Exists (TempFolder))
-				TempFolder = Path.Combine (TempFolder, "2");
+			try {
```

Comment:
Doing it in TearDown() may fail for the same reason as the other Delete() in TearDown(), open files on Windows. The first time SetUp() is called for this test it will delete any garbage that remains after a previous process which couldn't clean up after itself. The open files are closed and unlocked when the previous process finishes so the next time nunit is run the new process can delete the folder. It has to be done before the first test method is run though, hence in SetUp().